use-shopping-cart/src/index.js

```diff
@@ -129,37 +123,20 @@ export const useShoppingCart = () => {
   const handleCartHover = () => dispatch({ type: 'cart-hover' })
   const handleCloseCart = () => dispatch({ type: 'close-cart' })
 
-  async function redirectToCheckout(sessionId) {
-    if (stripe === null) throw new Error('Stripe is not defined')
-    const resolvedStripe = await Promise.resolve(stripe)
-
-    if (mode === 'client-only') {
-      // client-only checkout mode
-      const options = {
-        items: getCheckoutData.stripe(cart),
-        successUrl,
-        cancelUrl,
-        billingAddressCollection: billingAddressCollection
-          ? 'required'
-          : 'auto',
-        submitType: 'auto'
-      }
+  const redirectToCheckout = checkoutHandler(cart, {
+    modes: ['client-only', 'checkout-session'],
+    stripe(stripe, options) {
+      return stripe.redirectToCheckout(options)
+    }
+  })
 
-      if (allowedCountries?.length)
-        options.shippingAddressCollection = { allowedCountries }
-
-      const { error } = await resolvedStripe.redirectToCheckout(options)
-      if (error) return error
-    } else if (mode === 'checkout-session') {
-      // checkout-session mode
-      const { error } = await resolvedStripe.redirectToCheckout(sessionId)
-      if (error) return error
-    } else {
-      throw new Error(
-        `Invalid checkout mode '${mode}' was chosen. Valid options are 'client-only' and 'checkout-session'`
-      )
+  const checkoutSingleItem = checkoutHandler(cart, {
+    modes: ['client-only'],
+    stripe(stripe, options, { sku, quantity = 1 }) {
+      options.items = [{ sku, quantity }]
```

Comment:
This should fix using `checkoutSingleItem()` with the new Prices API from Stripe

```suggestion
      options.items = [{ price: sku, quantity }]
```